apps/api/src/index.ts

```diff
@@ -44,15 +88,37 @@ app.use(
 // This ensures we can properly rate limit by team ID
 app.use(rateLimiterMiddleware);
 
+const adminMiddleware = adminAuthMiddleware(services.teamManager);
+
+const accountController = makeAccountController(services);
+const adminController = makeAdminController(services);
+const competitionController = makeCompetitionController(services);
+const docsController = makeDocsController();
+const healthController = makeHealthController();
+const priceController = makePriceController(services);
+const publicController = makePublicController(services);
+const tradeController = makeTradeController(services);
+
+const accountRoutes = configureAccountRoutes(accountController);
+const adminRoutes = configureAdminRoutes(adminController, adminMiddleware);
+const adminSetupRoutes = configureAdminSetupRoutes(adminController);
```

Comment:
Made this new route for handling just the admin setup. It was confusing before how just one route in the the admin routes wasn't protected with auth, and that was achieved by the very specific order of when the middleware was registered with the router vs the routes. This keeps things simple where all the routes in the same file are treated with the same middleware.

Reply:
nice